src/hooks/useCanEditProfile.ts

```diff
@@ -1,24 +1,36 @@
 import { useEffect } from 'react';
-import { useNavigate } from 'react-router';
+import { useNavigate, useLocation } from 'react-router';
 
 import { LANDING_PAGE_ROUTE } from '~routes';
 import { getLastWallet } from '~utils/autoLogin';
 import { useAppContext } from '~hooks';
 
 const useCanEditProfile = () => {
-  const { user, userLoading, walletConnecting } = useAppContext();
+  const { user, userLoading, walletConnecting, updateUser, wallet } =
+    useAppContext();
 
   const navigate = useNavigate();
   // If there's a last wallet label, it means we're logged in.
   // If null, we're logged out and should be redirected to landing page.
   const lastWalletLabel = getLastWallet();
+  const { state: locationState } = useLocation();
+  const isRedirect = locationState?.isRedirect;
```

Comment:
The redirect is not working properly, as it always redirects (when going directly to the url), even if we have a wallet logged in.

I suspect it's because of the wallet not being available at the first render of the page